src/marqo/core/models/marqo_index.py

```diff
@@ -250,6 +259,8 @@ class MarqoIndex(ImmutableBaseModel, ABC):
     normalize_embeddings: bool
     text_preprocessing: TextPreProcessing
     image_preprocessing: ImagePreProcessing
+    video_preprocessing: Optional[VideoPreProcessing] = None
```

Comment:
If the user doesn't provide them then they should be populated in `class IndexSettings` so at the end of index creation they should have values.

I think we need to make them Optional now to avoid backwards compatibility issues, like if an older Index does not have these fields, they would error out or at least give a warning of missing field.

Reply:
I don't think we can do this since index_management deals with index settings right? And this one is an add docs object.

but I did push new code which:
- adds a new env var. Now we have two env vars for download threads: (1) image threadcount, and (2) media thread count. This is for depreciating the image one. Both can't be set explicitly at the same time or it will error.
- in tensor_search, a new method which selects the proper threadcount variable to use andsets the default media for languagebind to 5 threads.

Reply:
Good consideration on the backward compatibility issue! Thanks